### PR TITLE
Register

### PR DESCRIPTION
**Pull request by marijamitreska**

### CODE REVIEW

File: Project/demo/src/main/java/com/example/demo/model/User.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
+    package com.example.demo.model;
+
+    import javax.persistence.*;
+
+    @Entity
+    @Table(name = "userregister")
```
Comment (dejannastevski-armedia) [inline Project/demo/src/main/java/com/example/demo/model/User.java]: We need this table for saving users, so this table should be named "user" instead.

File: Project/demo/src/main/java/com/example/demo/model/authenticationController.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    import org.springframework.beans.factory.annotation.Autowired;
+    import org.springframework.stereotype.Controller;
+    import org.springframework.ui.Model;
+    import org.springframework.web.bind.annotation.GetMapping;
+    import org.springframework.web.bind.annotation.PostMapping;
+
+
+    @Controller
+    public class authenticationController
```
Comment (dejannastevski-armedia) [inline Project/demo/src/main/java/com/example/demo/model/authenticationController.java]: By convention class name always starts with capital letter. Object names and variable names start with small letter.

File: Project/demo/src/main/java/com/example/demo/model/authenticationController.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    {
+        @Autowired
+        private UserRepository userRepo;
+
+        @GetMapping(" ")
```
Comment (dejannastevski-armedia) [inline Project/demo/src/main/java/com/example/demo/model/authenticationController.java]: Remove the empty space in the @GetMapping if you want that to be default mapping value.

File: Project/demo/src/main/java/com/example/demo/model/authenticationController.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+            return "index";
+        }
+        @GetMapping("/register")
+        public String showRegistrationForm(Model model)
+        {
+            model.addAttribute("user", new User());
+
+            return "signup_form";
+        }
+        @PostMapping("/process_register")
```
Comment (dejannastevski-armedia) [inline Project/demo/src/main/java/com/example/demo/model/authenticationController.java]: Change to "/process-register" instead.

File: Project/demo/src/main/java/com/example/demo/model/UserRepository.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
+    package com.example.demo.model;
```
Comment (aleksandar-bujaroski-armedia) [inline Project/demo/src/main/java/com/example/demo/model/UserRepository.java]: UserRepository should be in separate repository directory in order to follow MVC model

File: Project/demo/src/main/java/com/example/demo/model/authenticationController.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
+    package com.example.demo.model;
```
Comment (aleksandar-bujaroski-armedia) [inline Project/demo/src/main/java/com/example/demo/model/authenticationController.java]: AuthenticationController should be in separate controller directory in order to follow MVC model

File: Project/demo/src/main/java/com/example/demo/model/authenticationController.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    import org.springframework.ui.Model;
+    import org.springframework.web.bind.annotation.GetMapping;
+    import org.springframework.web.bind.annotation.PostMapping;
+
+
+    @Controller
+    public class authenticationController
+    {
+        @Autowired
+        private UserRepository userRepo;
```
Comment (aleksandar-bujaroski-armedia) [inline Project/demo/src/main/java/com/example/demo/model/authenticationController.java]: Refactor the code and extract business logic to Service

File: Project/demo/src/test/java/com/example/demo/UserRepositoryTest.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
+    package com.example.demo;
```
Comment (aleksandar-bujaroski-armedia) [inline Project/demo/src/test/java/com/example/demo/UserRepositoryTest.java]: Add this test code to separate directory called test

**Review by aleksandar-bujaroski-armedia — CHANGES_REQUESTED** [review]

File: Project/demo/pom.xml [hunk 9]
```diff
@@ -19,14 +19,61 @@
...
 		</dependency>
+		<dependency>
+		<groupId>org.webjars</groupId>
+		<artifactId>jquery</artifactId>
+		<version>3.4.1</version>
+	</dependency>
+	<dependency>
+		<groupId>org.webjars</groupId>
+		<artifactId>bootstrap</artifactId>
+		<version>4.3.1</version>
```
Comment (aleksandar-bujaroski-armedia) [inline Project/demo/pom.xml]: Extract version number <version>4.3.1</version> to variable at the top of the file:
Example:
```
   <properties>
        <bootstrap.version>4.3.1</bootstrap.version>
   </properties>

```
Then use it in this way:
`<version>${bootstrap.version}</version>`

**Review by aleksandar-bujaroski-armedia — CHANGES_REQUESTED** [review]

File: Project/demo/src/main/java/com/example/demo/controller/AuthenticationController.java [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+        return "signup_form";
+    }
+
+    @PostMapping("/register/redirect-to-login")
+    public String loginForm(User user, Model model)
+    {
+        String res = authenticationService.validateUser(user);
+        if(res.length() == 0)
+        {
+
```
Comment (dejannastevski-armedia) [inline Project/demo/src/main/java/com/example/demo/controller/AuthenticationController.java:37]: Remove these empty lines.

File: Project/demo/src/main/java/com/example/demo/controller/HelloController.java [hunk 11]
```diff
@@ -1,14 +1,14 @@
...
-        @RestController
-        public class HelloController
-        {
-            @RequestMapping("/hello")
-            public String helloWorld()
-            {
-                return "Hello World";
-            }
-        }
+//        package com.example.demo.controller;
```
Comment (dejannastevski-armedia) [inline Project/demo/src/main/java/com/example/demo/controller/HelloController.java:1]: remove commented code.

**Pull request merged**